lib/rubocop/cop/lint/unreachable_code.rb

```diff
@@ -88,6 +106,33 @@ def check_case(node)
 
           branches.all? { |branch| branch.body && flow_expression?(branch.body) }
         end
+
+        def handle_def(node)
+          @redefined << node.method_name
+          false
+        end
+
+        def instance_eval_block?(node)
+          node.block_type? && node.method?(:instance_eval)
+        end
+
+        def report_on_flow_command?(node)
+          return true unless node.send_type?
+
+          # By the contract of this function, this case means that
+          # the method is called on `Kernel` in which case we
+          # always want to report a warning.
+          return true if node.children[0] != nil
```

Comment:
Since `node` is guaranteed to be a `send` node here, we can use node methods here.

```suggestion
          return true if node.receiver
```